dashboard/legacy/middleware/helpers/projects.rb

```diff
@@ -457,4 +464,27 @@ def project_type_from_merged_row(row)
     # Report these as 'unknown'.
     'unknown'
   end
+
+  # Temporarily logging rather than erroring in order
+  # to confirm that only valid thumbnail URLs are present in existing projects.
+  # DCDO flag in place to choose when to enable/disable logging (off by default).
+  def validate_thumbnail_url(channel_id, thumbnail_url)
+    return true unless thumbnail_url
+
+    # valid thumbnail URLs should be of the format:
+    # /v3/files/<channel_id>/.metadata/thumbnail.png
+    # I observed thumbnail URLs of remixed projects having having the channel ID of the parent project,
+    # so we assert on the start/end of the URL
+    valid_thumbnail_url = thumbnail_url.start_with?('/v3/files/') && thumbnail_url.end_with?('.metadata/thumbnail.png')
+    if !valid_thumbnail_url && DCDO.get('log_thumbnail_url_validation', false)
+      # raise ValidationError
+      Honeybadger.notify(
+        error_class: 'Project::ValidationError',
```

Comment:
Is this why you created a new error? so we can have it mapped on honeybadger

I am curious, because I think it is a good idea. I just wonder how it could be useful in honeybadger or it will be consumed compared to an already existing error. 

This is me trying to learn here

Reply:
I don't have a ton of experience here so I wouldn't take too much from my process, but I guess I was thinking that there wasn't a good existing error type to represent the type of error I'm trying to track here. I saw the existing pattern used to subclass a Sinatra error (`NotFound`) and followed it.

That said, I think the "right" place for this would be as validation in Rails as part of our `Project` [model](https://github.com/code-dot-org/code-dot-org/blob/staging/dashboard/app/models/project.rb), but realized working on this that we unfortunately do not actually use Rails (ie, we don't instantiate a Project, check validation before saving, etc -- you probably know this :)). I checked in with a couple folks on we could migrate these routes to Rails and do validation there, and it sounds like that work is possible now, time permitting (which I think would be a lot cleaner than what I have here).